packages/doc/docs/quickstart.mdx

```diff
@@ -0,0 +1,444 @@
+
+    ```sql
+    SELECT * FROM customers
+    WHERE UPPER(first_name) = {{ context.params.name | upper }}
+    LIMIT 1
+    ```
+
+    Now, test with some names in lower case:
+
+    - [http://localhost:3000/api/customer?name=kendell](http://localhost:3000/api/customer?name=kendell)
+
+3.  Throw error when user not found. We can let our API better by showing an accurate error message.
+    First, we need to execute an extra query to know whether the user is in our database:
+
+    ```sql
+    {% req user %}
+    SELECT COUNT(\*) AS count FROM customers
+    WHERE UPPER(first_name) = {{ context.params.name | upper }}
+    {% endreq %}
+
+    SELECT * FROM customers
+    WHERE UPPER(first_name) = {{ context.params.name | upper }}
+    LIMIT 1
+    ```
+
+    The block `{% req user %} ... {% endreq %}` is a [query block](./api-building/sql-builder) which tells VulcanSQL that we want to save the query result to `user` variable instead of outputting as responses.
+
+    Now we can get the result of this query and throw an error when the result equals `0`.
+
+    ```sql
+    {% req user %}
+    SELECT COUNT(*) AS count FROM customers
+    WHERE UPPER(first_name) = {{ context.params.name | upper }}
+    {% endreq %}
+
+    {% if user.value()[0].count == 0 %}
+      {% error "CUSTOMER_NOT_FOUND" %}
+    {% endif %}
+
+    SELECT * FROM customers
+    WHERE UPPER(first_name) = {{ context.params.name | upper }}
+    LIMIT 1
+    ```
+
+    We used the if expression to throw an error when the result equals `0`, VulcanSQL server will stop executing and respond immediately when meeting a {% error %} tag, `"CUSTOMER_NOT_FOUND"` is the error code we want to throw.
+
+    :::info
+    You can add more information about your errors, e.g. description, HTTP code …etc. Please check [Error response](./api-building/error-response)
+
+    :::
+
+    You can test with some invalid names:
+
+    - [http://localhost:3000/api/customer?name=some-invalid-name](http://localhost:3000/api/customer?name=some-invalid-name)
+
+4.  Throw an error when the name is ambiguous. We noticed that some customers have the same first name, let's figure them out and throw an error.
+    We'll need to use the user's count twice, in order the reuse the result, we need to save the result first.
+
+    ```sql
+    {% req user %}
+    SELECT COUNT(\*) AS count FROM customers
+    WHERE UPPER(first_name) = {{ context.params.name | upper }}
+    {% endreq %}
+
+    {% set userCount = user.value()[0].count %}
+
+    {% if userCount == 0 %}
+      {% error "CUSTOMER_NOT_FOUND" %}
+    {% endif %}
+
+    SELECT * FROM customers
+    WHERE UPPER(first_name) = {{ context.params.name | upper }}
+    LIMIT 1
+    ```
+
+    {% set %} tag saved the result from the right side like most programming languages: `var someVar = someVal`, in this example, we saved the query result into `userCount` variable.
```

Comment:
Can we add the quote to be `{% set %}` ?

Reply:
Sure.